### ISSUE TITLE
How do I put pagination bar on top of grid

### ISSUE THREAD

**Issue by euvin3582** [angular-ui/ui-grid#5193]
Is there a built in feature that allows me to have pagination on top and bottom of the grid?

**Comment by angular-git-user** [comment]
You will need to detach and append in a div above.. I have used this for my project..

or just create youre own pagination.///

`<div id="top-pagination" class="ui-grid-pager-control">
                                <button type="button" class="ui-grid-pager-first" ng-click="gridCtrl.gridApi1.pagination.seek(1)" title="Page to first" aria-label="Page to first">
                                    <div class="first-triangle"><div class="first-bar"></div></div>
                                </button>
                                <button type="button" class="ui-grid-pager-previous" ng-click="gridCtrl.gridApi1.pagination.previousPage()" ng-disabled="cantPageBackward()" title="Page back" aria-label="Page back">
                                    <div class="first-triangle prev-triangle"></div>
                                </button>
                                <span>{{gridCtrl.gridApi1.pagination.getPage()}}/{{gridCtrl.gridApi1.pagination.getTotalPages()}}</span>
                                <button type="button" class="ui-grid-pager-next" ng-click="gridCtrl.gridApi1.pagination.nextPage()" title="Page forward" aria-label="Page forward">
                                    <div class="last-triangle next-triangle"></div>
                                </button>
                                <button type="button" class="ui-grid-pager-last" ng-click="gridCtrl.gridApi1.pagination.seek(gridCtrl.gridApi1.pagination.getTotalPages())" title="Page to last" aria-label="Page to last">
                                    <div class="last-triangle"><div class="last-bar"></div></div>
                                </button>
                            </div>`

**Comment by angular-git-user** [comment]
sorry for the formatting

**Comment by angular-git-user** [comment]
and for detach append use this code
`var bottomPagination = $("#grid1").find(".ui-grid-pager-container").detach();`
`            ('#youreDivId').append(bottomPagination );`

**Comment by Gajendrasinh** [comment]
how can i pass page size in 

<div id="top-pagination" class="ui-grid-pager-control">
   <button type="button" class="ui-grid-pager-first" ng-click="gridCtrl.gridApi1.pagination.seek(1)" title="Page to first" aria-label="Page to first">
      <div class="first-triangle">
         <div class="first-bar"></div>
      </div>
   </button>
   <button type="button" class="ui-grid-pager-previous" ng-click="gridCtrl.gridApi1.pagination.previousPage()" ng-disabled="cantPageBackward()" title="Page back" aria-label="Page back">
      <div class="first-triangle prev-triangle"></div>
   </button>
   <span>{{gridCtrl.gridApi1.pagination.getPage()}}/{{gridCtrl.gridApi1.pagination.getTotalPages()}}</span> 
   <button type="button" class="ui-grid-pager-next" ng-click="gridCtrl.gridApi1.pagination.nextPage()" title="Page forward" aria-label="Page forward">
      <div class="last-triangle next-triangle"></div>
   </button>
   <button type="button" class="ui-grid-pager-last" ng-click="gridCtrl.gridApi1.pagination.seek(gridCtrl.gridApi1.pagination.getTotalPages())" title="Page to last" aria-label="Page to last">
      <div class="last-triangle">
         <div class="last-bar"></div>
      </div>
   </button>
</div>

this custom pagination

**Comment by puneethravi2** [comment]
this can be achieved by css, as ag-grid does not supports the positioning of the pagination component
`.ag-theme-balham {
  background-color: transparent;
  margin-top: -30px;
}

.ag-paging-panel > span {
  color: black !important;
}

.ag-root-wrapper {
  position: relative !important;
}

.ag-root-wrapper-body {
  margin-top: 30px;
}

.ag-paging-panel {
  position: absolute !important;
  top: 0 !important;
  right: 0 !important;
  border-top: none !important;
}`
